include/xsimd/arch/xsimd_wasm.hpp

```diff
@@ -415,6 +427,59 @@ namespace xsimd
             return wasm_f64x2_eq(self, other);
         }
 
+        // fast_cast
+        namespace detail
+        {
+            template <class A>
+            inline batch<float, A> fast_cast(batch<int32_t, A> const& self, batch<float, A> const&, requires_arch<wasm>) noexcept
+            {
+                return wasm_f32x4_convert_i32x4(self);
+            }
+
+            template <class A>
+            inline batch<float, A> fast_cast(batch<uint32_t, A> const& v, batch<float, A> const&, requires_arch<wasm>) noexcept
+            {
```

Comment:
This is exactly the same operation we use for sse2... I think we should make it generic, I'll give it a try.